File: master/DataGridViewEx-VBNET/ColumnInfo.vb
''' <summary>
''' Class used to save column info for saving/restoring configuration
''' </summary>
''' <url>http://www.extensionmethod.net/csharp/datagridview/load-save-configuration</url>
<Serializable>
Public NotInheritable Class ColumnInfo

  Private m_Name As String
  Public Property Name() As String
    Get
      Return m_Name
    End Get
    Set
      m_Name = Value
    End Set
  End Property

  Private m_DisplayIndex As Integer
  Public Property DisplayIndex() As Integer
    Get
      Return m_DisplayIndex
    End Get
    Set
      m_DisplayIndex = Value
    End Set
  End Property

  Private m_Width As Integer
  Public Property Width() As Integer
    Get
      Return m_Width
    End Get
    Set
      m_Width = Value
    End Set
  End Property

  Private m_Visible As Boolean
  Public Property Visible() As Boolean
    Get
      Return m_Visible
    End Get
    Set
      m_Visible = Value
    End Set
  End Property

End Class

File: master/DataGridViewEx/ColumnInfo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace DataGridViewEx
{
  /// <summary>
  /// Class used to save column info for saving/restoring configuration
  /// </summary>
  /// <url>http://www.extensionmethod.net/csharp/datagridview/load-save-configuration</url>
  [Serializable()]
  public sealed class ColumnInfo
  {
    public string Name { get; set; }

    public int DisplayIndex { get; set; }

    public int Width { get; set; }

    public bool Visible { get; set; }

  }

}
